vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -439,6 +458,51 @@ def _nixl_handshake_listener(metadata: NixlAgentMetadata,
                         "Connection listener got unexpected message %s", msg)
                 sock.send_multipart((identity, b"", encoded_data))
 
+    def _start_handshake(self, engine_id: str, host: str, port: int):
+        """Start handshake using ThreadPoolExecutor.
+        
+        This method is non-blocking and submits `_nixl_handshake` to the
+        background thread pool.
+        """
+        if engine_id in self._handshake_futures:
+            return
+
+        logger.debug("Starting handshake for engine %s", engine_id)
+        future = self._executor.submit(self._nixl_handshake, host, port)
+        self._handshake_futures[engine_id] = future
+
+        # Set up callback to handle completion
+        def on_handshake_complete(fut: Future):
+            logger.debug("Handshake callback triggered for engine %s",
+                         engine_id)
+            with self._lock:
+                if engine_id in self._handshake_futures:
+                    # Clean up futures. In case of failure, requests will remain
+                    # pending and be reported to scheduler for retry.
+                    del self._handshake_futures[engine_id]
+                try:
+                    fut.result()  # This will raise if the handshake failed
+                    logger.debug("Handshake succeeded for engine %s", engine_id)
+                    completed_reqs = self._pending_requests.pop(engine_id, None)
```

Comment:
Why would the pending_requests be completed only when the handshake is done? We do not communicate pending requests from P instances in the handshake thread right?